### ISSUE TITLE
Documentation: What are appropriate values for try_keep_nr?

### ISSUE THREAD

**Issue by nickmitchko** [Mihaiii/llm_steer#3]
What range of values for try_keep_nr are valid?

Is there a metric in a transformers model config that correlates with the upper bound of this number?

**Comment by Mihaiii** [comment]
@nickmitchko 
I added back the docstrings that were removed in #2 .

For "try_keep_nr", the docstring is: 
```
try_keep_nr: This is used in advanced usage and determines the number of rows of the initial matrix to be kept. The param is used for experimenting. Leave to default value for best usage.
```

I usually leave the default value (meaning `None`). Empirically, the app works best if I don't change it, but I wanted to allow the user the possibility to play with it.

The `try_keep_nr` parameter values are between 0 (meaning the steering vector is applied on the whole matrix) and the number of tokens that form the text you pass when add a steering vector. Be aware that, by default, I inject a BOS token, which changes the number of tokens (I also have an option to remove this behavior: setting the `exclude_bos_token` param to `True`)

![image](https://github.com/Mihaiii/llm_steer/assets/8951553/c2ac4522-195b-45c5-816e-6555f6653ae1)

^ In the above example `try_keep_nr` can be between 0 and 1 because we have 2 tokens for the text "cat": the BOS token and the actual token that the tokenizer provides. For a value of 2 or higher, the steering vector would not be applied - I decided to raise an error in this case (why add a steering vector with if you have no intention to apply it? This certainly wasn't the intention of the user when they added the steering vector.).

Let me know if you have more questions.
